GitCommands/CommitMessageManager.cs

```diff
@@ -100,7 +106,20 @@ public string MergeOrCommitMessage
                 }
 
                 var (file, exists) = GetMergeOrCommitMessagePath();
-                return exists ? _fileSystem.File.ReadAllText(file, _commitEncoding) : string.Empty;
+                string result;
+                try
+                {
+                    result = exists ? _fileSystem.File.ReadAllText(file, _commitEncoding) : string.Empty;
+                }
+                catch (UnauthorizedAccessException)
```

Comment:
Could be any Exception - not relevant to crash due to this

Reply:
Yes, I'd prefer to catch all exceptions.

Reply:
updated